### DISCUSSION TITLE
Custom Preset for mixcr-4

### DISCUSSION THREAD

**Discussion by msadv** [milaboratory/mixcr#1029] [Q&A]
Could someone please assist me with finding documentation on creating a custom preset for mixcr-4? I have been searching for a preset that meets my requirements, but have not been successful thus far. I understand that a preset is a required argument in mixcr-4, so any help in creating a custom preset would be greatly appreciated.

**Comment by Alex-Davydov** [comment]
Hi,
If you provide us with the library structure (positions of UMI, CELL barcodes etc), we will prepare a preset for your protocol.
Best,
Alex

**Reply by msadv** [reply]
Hi,
I am sharing an image with different possibilities of the preset I am looking for. We have adapters that need to be used as well. To visualize that the adapter could be anywhere is why there are multiple examples given below. Please let me know if you have any questions.
 
<img width="786" alt="Screen Shot 2023-03-08 at 9 03 23 PM" src="https://user-images.githubusercontent.com/25792512/223964135-b26bb089-1929-4453-9c89-f133e31bce6c.png">

**Comment by Alex-Davydov** [comment]
Hi,
Unfortunately now MiXCR doesn't support incomplete rearrangements. This functionality will be added in a couple of months.
Are this scheme related to IGK incomplete rearrangements and C(intron) is a Intron-RSS sequence? If not, could you explain what is C(intron), we need to add it to the MiXCR reference.
Best,
Alex

**Reply by msadv** [reply]
It looks like there was a misunderstanding. By constant intron, it is a reference to the J-C intron, which is an option in mixcr ([https://mixcr.com/mixcr/reference/overview-mixins-list/#alignment-mix-in-options](https://urldefense.com/v3/__https://mixcr.com/mixcr/reference/overview-mixins-list/*alignment-mix-in-options__;Iw!!LIr3w8kk_Xxm!qL5FsrjlXZns2YGxWUtqRTQMoEv1oPlUP4tN3yCZXYWHEiU5EdDBg45M-EaiAun2cZ-_rLed7PMDM-zRCA$)) and also it is a complete rearrangement.

**Comment by Alex-Davydov** [comment]
Got it, and you need to assemble full-length VDJ from these fragments? Could you provide me with the adapter sequence? Does it contain any tags (eg UMI, cell barcode etc)?

**Reply by msadv** [reply]
No, we can't assemble the full sequence if the fragment is a partial sequence. Below are the adapter sequences.
Forward adapter (TruSeq Read1)       5’-ACACTCTTTCCCTACACGACGCTCTTCCGATCT-3’
Reverse adapter (TruSeq Read2)        5’- GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT-3’
There is no UMI or cell barcode.

**Reply by Alex-Davydov** [reply]
Hi, it seems that the preset for RNAseq should work with your data.
`mixcr analyze rnaseq-full-length --species <species> R1.fastq(.gz) R2.fastq(.gz) result/`

**Reply by msadv** [reply]
Hi, 
So I am using a custom reference library which is a bisulphite converted (G->A) IMGT library for this and it does not give the output as expected with no tsv files being generated. I am attaching the library files below. Please let me know if you have any input on this.

[my_library.zip](https://github.com/milaboratory/mixcr/files/10987240/my_library.zip)

**Comment by Alex-Davydov** [comment]
Hi,
For normal alignment you need to specify positions of CDR3Begin in V gene and CDR3End (FR4Begin) in J gene. If you want to extract other regions (frameworks and cdrs) you also need to specify their positions. Here you can find manual about MiXCR library construction from fasta files https://mixcr.com/mixcr/guides/create-custom-library/

**Reply by msadv** [reply]
The link you have posted gives me a Page not found error.

**Reply by PoslavskySV** [reply]
Please, use this link: https://mixcr.com/mixcr/guides/create-custom-library/
